### PR TITLE
Added Support to generate different multiplicity relations

### PR DESCRIPTION
**Pull request by tedhtchang**
Let users specify mulitiplicity (ONE2ONE, MANY2ONE,ONE2MANY,SIMPLE,MULTI)
and self reference in the csv config json. Generate edge files with respect
to multiplicity.

Resolves: #20, #21
Signed-off-by: ted chang <htchang@us.ibm.com>

### CODE REVIEW

File: src/com/ibm/janusgraph/utils/generator/CSVGenerator.java [hunk 1]
```diff
@@ -193,8 +173,7 @@ public void writeEdgeCSVs(EdgeTypeBean type, String outputDirectory ){
                 System.out.println("Generated edge file: "+ csvFile);
             }
         } catch (Exception e) {
-            // TODO Auto-generated catch block
-            throw new RuntimeException(e.toString());
+            e.printStackTrace();
```
Comment (chinhuang007) [inline src/com/ibm/janusgraph/utils/generator/CSVGenerator.java]: Do you want to hide the exception? Maybe consider to throw after you print the stack trace?

File: src/com/ibm/janusgraph/utils/generator/IdStore.java [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+import java.util.Map;
+
+import org.apache.commons.lang3.RandomUtils;
+import com.ibm.janusgraph.utils.generator.bean.CSVIdBean;
+import com.ibm.janusgraph.utils.generator.bean.RelationBean;
+
+public class IdStore {
+    private CSVIdBean idBean;
+    private SimpleRelationStore simpleStore = null;
+    private RelationStore generalstore = null;
```
Comment (yhwang) [inline src/com/ibm/janusgraph/utils/generator/IdStore.java:32]: Is a IDStore for one edge? if that's the case, I guess you could only have one RelationStore? Is it possible to abstract SimpleRelationStore and RelationStore to one single interface?

File: src/com/ibm/janusgraph/utils/generator/IdStore.java [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+        private Map<Integer, HashSet<Integer>> store = new HashMap<Integer, HashSet<Integer>>();
+        private RelationBean relation = null;
+        private List<Integer> excl_left_id;
+        public SimpleRelationStore(RelationBean relation) {
+            this.relation = relation;
+            this.excl_left_id = new ArrayList<Integer>();
+        }
+        public boolean put(int left, int right) {
+            if (! store.containsKey(left)) {
+                store.put(left, new HashSet<Integer>());
```
Comment (yhwang) [inline src/com/ibm/janusgraph/utils/generator/IdStore.java:308]:
```
HashSet<Integer> newSet = new HashSet<Integer>();
store.put(left, newSet);
if (!relation.selfRef) {
    return newSet.add(left); //should you return here? that's my guess
}
```

And I think the `selfRef` logic should also apply to `else` case. right?

File: src/com/ibm/janusgraph/utils/generator/IdStore.java [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+            return store.get(left).add(right);
+        }
+        public List<Integer> getIds(int left) {
+            if (! store.containsKey(left)) {
+                store.put(left, new HashSet<Integer>());
+                if (!relation.selfRef) {
+                    store.get(left).add(left);
+                }
+            }
+            List<Integer> a = new ArrayList<Integer>(store.get(left));
```
Comment (yhwang) [inline src/com/ibm/janusgraph/utils/generator/IdStore.java:322]: maybe you could directly use SortedSet (TreeSet)

**Review by yhwang — CHANGES_REQUESTED** [review]

**Review by yhwang — APPROVED** [review]
per our discussion, we can improve it later. approve it for now

**Pull request merged**